Navigator cleanup: remove IDLE setpoint type

mbjd opened the pull request:
Remake of https://github.com/PX4/PX4-Autopilot/pull/23704, removing `position_setpoint_s::SETPOINT_TYPE_IDLE` and close relatives, namely:
 - `NAV_CMD_IDLE`
 - `MissionBlock::set_idle_item`
 - `WaypointType::idle` from `FlightTaskAuto`, which mirrors the position setpoint types defined in `PositionSetpoint.msg`

### Why should we remove it? 

Idle position setpoints / mission items currently serve two functions, both of which are already provided elsewhere:
 1. When landed, in various places make the _current_ position setpoint idle or give an idle mission item. 
      - Even without the idle setpoint these landings work as expected.
 2. When using `reset_position_setpoint`, the setpoint is marked invalid and also given the idle type. This is often done to prevent line-following between previous-current setpoints.
      - The idle type is only checked explicitly in very few places, all of which allow alternatives which in my opinion are clearer:
         - in the rover modules we instead check for setpoint validity
         - in the terrain collision avoidance logic in `mission_block` we rely on the landed state directly, rather than the idle setpoint type which in that context is only a proxy of the landed state. 

Beyond this not very clearly defined functionality, they also present a simple path to a fatal mistake -- if the current position setpoint ever is of type `IDLE` in fixed-wing flight, then `FixedWingModeManager::set_control_mode_current` will switch to `FW_POSCTRL_MODE_AUTO`, which will then happily command level attitude and zero thrust, completely neglecting TECS. Depending on the airframe this will result in an uncontrolled descent or stall.

### Where is it currently used and why doesn't removing it break things? 

This is an exhaustive list of the ways in which an idle position setpoint currently enters the system, along with my reasoning why removing it should be unproblematic:

 - given exactly when the land detector says we have landed:                                                                                          
    - in `land.cpp` via `set_idle_item`                                                                                                 
    - in `rtl_direct.cpp` via `set_idle_item`                                                                              
    - in `mission_base.cpp` via mission item with `.nav_cmd = NAV_CMD_IDLE`
    - in `takeoff.cpp`, `on_active()`, via mission item with `.nav_cmd = NAV_CMD_IDLE` 
      - these four cases are SITL tested and work the same as previously
- as initial value                                                                                                                                                
    - Initialisation of _type and _type_previous in FlightTaskAuto.hpp with WaypointType::idle. Note this only concerns the mirrored version within `FlightTaskAuto` and does not come from a position setpoint itself.
 - Through `reset_position_setpoint` and `reset_triplets`, from which are used quite often (see search results below :mag:)
   - I have tested all of these cases in SITL, verifying basic functionality. 
   - I am also quite convinced this cleanup _should_ not break anything related to reset / invalidated setpoints, as there is currently not a single place where setpoint (in)validity is checked by comparing the type to idle -- rather, the `valid` flag or `PX4_ISFINITE` of the relevant entries are used.  

<details>
  <summary>:mag: Search results for  reset_position_setpoint and reset_triplets</summary>

```
$ grep -r reset_position_setpoint ./src                                                   
./src/modules/navigator/vtol_takeoff.cpp:				_navigator->reset_position_setpoint(reposition_triplet->previous);         
./src/modules/navigator/vtol_takeoff.cpp:				_navigator->reset_position_setpoint(reposition_triplet->current);          
./src/modules/navigator/vtol_takeoff.cpp:				_navigator->reset_position_setpoint(reposition_triplet->next);             
./src/modules/navigator/navigator_main.cpp:	reset_position_setpoint(_pos_sp_triplet.previous);                                     
./src/modules/navigator/navigator_main.cpp:	reset_position_setpoint(_pos_sp_triplet.current);                                      
./src/modules/navigator/navigator_main.cpp:	reset_position_setpoint(_pos_sp_triplet.next);                                         
./src/modules/navigator/navigator_main.cpp:void Navigator::reset_position_setpoint(position_setpoint_s &sp)                        
./src/modules/navigator/mission_base.cpp:			_navigator->reset_position_setpoint(pos_sp_triplet->previous);                 
./src/modules/navigator/mission_base.cpp:			_navigator->reset_position_setpoint(pos_sp_triplet->previous);                 
./src/modules/navigator/takeoff.cpp:					_navigator->reset_position_setpoint(reposition_triplet->previous);         
./src/modules/navigator/takeoff.cpp:					_navigator->reset_position_setpoint(reposition_triplet->current);          
./src/modules/navigator/takeoff.cpp:					_navigator->reset_position_setpoint(reposition_triplet->next);             
./src/modules/navigator/rtl_direct.cpp:			_navigator->reset_position_setpoint(pos_sp_triplet->previous);                     
./src/modules/navigator/navigator.h:	void reset_position_setpoint(position_setpoint_s &sp);                                     
./src/modules/navigator/rtl_mission_fast_reverse.cpp:				_navigator->reset_position_setpoint(pos_sp_triplet->previous); 
./src/modules/navigator/rtl_mission_fast_reverse.cpp:			_navigator->reset_position_setpoint(pos_sp_triplet->previous);     
./src/modules/navigator/rtl_mission_fast_reverse.cpp:			_navigator->reset_position_setpoint(pos_sp_triplet->next);         
$ grep -r reset_triplets ./src                                                             
./src/modules/navigator/navigator_main.cpp:	reset_triplets();                                                                      
./src/modules/navigator/navigator_main.cpp:				reset_triplets();                                                          
./src/modules/navigator/navigator_main.cpp:			reset_triplets();                                                              
./src/modules/navigator/navigator_main.cpp:void Navigator::reset_triplets()                                                        
./src/modules/navigator/mission.cpp:			_navigator->reset_triplets();                                                      
./src/modules/navigator/mission_base.cpp:		_navigator->reset_triplets();                                                      
./src/modules/navigator/navigator.h:	void reset_triplets();                                                                     
```
</details>


### Please review!

Despite sim testing everything, I cannot completely rule out changing the behaviour in a subtle, possibly detrimental way - I am looking forward to comments assessing this risk (maybe I requested reviews a bit too generously - only review if you feel like you can give a good review without wasting hours :) 

P.S. the diff in `loiter.cpp` looks less scary when viewed with `?w=1` (web) or `-w` (terminal)

mbjd committed:
mecanum: replace previous waypoint type == idle check with validity check

mbjd committed:
ackermann: replace previous waypoint type == idle check with validity check

mbjd committed:
differential: replace previous waypoint type == idle check with validity check

mbjd committed:
mission_block: replace idle with landed check

which is the alternative solution mentioned in the original PR: https://github.com/PX4/PX4-Autopilot/pull/23674

mbjd committed:
FixedWingModeManager: give idle throttle if landed

 - except in takeoff where that is nonsensical
 - the other control_auto* and the two control_manual* functions already
   have such logic

dakejahl reviewed (dismissed):
I'm going to have to take your word for it. If you want to keep going on navigator I'm here for it. I've never been able to wrap my head around the logic in this module.

mbjd committed:
remove SETPOINT_TYPE_IDLE everywhere

along with NAV_CMD_IDLE, set_idle_item, and WaypointType::idle which are
basically the same

mbjd commented:
rebased on main & bump to wake up reviewers :) despite the large investigation I did last time I cannot rule out nasty regressions. I would appreciate some expert intuition in pointing to possible failure cases so I can look at and possibly address them.

mbjd committed:
FixedWingModeManager: remove IDLE handling that moved in rebase

dakejahl commented:
should we get this in?
